projects/storefrontlib/cms-components/content/tab/tab.component.ts

```diff
@@ -0,0 +1,178 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ChangeDetectionStrategy,
+  Component,
+  Input,
+  OnInit,
+  QueryList,
+  ViewChildren,
+} from '@angular/core';
+import { BreakpointService } from '../../../layout/breakpoint';
+import { BehaviorSubject, Observable, of } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { Tab, TabConfig, TAB_MODE } from './tab.model';
+import { wrapIntoBounds } from './tab.utils';
+
+@Component({
+  selector: 'cx-tab',
+  templateUrl: './tab.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TabComponent implements OnInit {
+  @Input() tabs: Tab[];
+  @Input() config: TabConfig | any;
+
+  TAB_MODE = TAB_MODE;
+
+  openTabs$: BehaviorSubject<number[]>;
```

Comment:
Need to consider benefits of using these observables to control the tab component. Why? Main reason I can think of is to reduce complexity. So far:
- I tried to use static variables of `isOpen` and attach them to the `Tab[]` object but this is NOT responsive in practice.
- I could try to leverage the `openTabs`  property in `TabConfig`. I may still need an observable in the end for responsiveness though.